src/core/src/energy/global/wolf.rs

```diff
@@ -216,10 +220,13 @@ impl GlobalPotential for Wolf {
 
     fn virial(&self, system: &System) -> Matrix3 {
         let natoms = system.size();
-        let mut res = Matrix3::zero();
-        for i in 0..natoms {
+
+        (0..natoms).into_par_iter().map(|i| {
+
             let qi = system[i].charge;
-            if qi == 0.0 {continue}
+            if qi == 0.0 { return Matrix3::zero(); }
+            let mut virial = Matrix3::zero();
```

Comment:
Should we call all local variables `local_***`? I don't care, as long as it is consistent in all the code (Ewald uses local_virial)